File: server/dotnet/sdk-dotnet-example/Presentation/Mappers/PaymentDetailsMapper.cs
using Business.DTOs.GetPaymentDetails;
using Presentation.Models.GetPaymentDetails;

namespace Presentation.Mappers;

public static class PaymentDetailsMapper
{
    public static PaymentDetailsResponse Map(PaymentDetailsResponseDto response)
    {
        return new()
        {
            Id = response.Id,
            Status = response.Status,
            StatusOutput = response.StatusOutput,
            Operations = response.Operations,
            PaymentOutput = response.PaymentOutput,
            HostedCheckoutSpecificOutput = response.HostedCheckoutSpecificOutput
        };
    }
}

File: server/java/src/main/java/com/onlinepayments/example/sdk/mappers/PaymentDetailsMapper.java
package com.onlinepayments.example.sdk.mappers;

import com.onlinepayments.domain.PaymentDetailsResponse;
import com.onlinepayments.example.service.dtos.paymentdetails.PaymentDetailsResponseDto;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface PaymentDetailsMapper {
    PaymentDetailsResponseDto toResponseDto(PaymentDetailsResponse response);
}
